fecgov/fec-cms
# Feature/4171 o10n preconnect to required origins

## Summary

- Resolves #4171 

Lighthouse flagged that we could help our performance by preconnecting to third-party sources

## Impacted areas of the application

Created two new files: an html and jinja file for the `preconnect` and `dns-prefetch`
Updated three templates to include the new files for production

## Screenshots

No visible changes, but here's the code we're looking for:
![image](https://user-images.githubusercontent.com/26720877/110037207-14faea80-7d0c-11eb-9256-10c6021ebf71.png)



## Related PRs

None

## How to test

- Pull branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Check the homepage
  - In fec/fec/templates/home_base.html
    - edit line 7 to be ``
    - edit line 9 to be ``
    - or just delete them both together.
  - Load the [homepage](http://127.0.0.1:8000)
  - View the page source or "inspect" the elements, your preference
  - Locate the code in the screenshots above (or cmd-f for "preconnect")
  - Should see the code shown above
- Check the base template
  - In fec/fec/templates/base.html
    - edit line 7 to be ``
    - edit line 9 to be ``
    - or just delete them both together.
  - Load a [page](http://127.0.0.1:8000/about/) that extends the base template
  - View the page source or "inspect" the elements, your preference
  - Locate the code in the screenshots above (or cmd-f for "preconnect")
  - Should see the code shown above
  {% include 'partials/meta-tags-preconnects.jinja' %}
  {# {% endif %} #}
- Check the main template
  - In fec/data/templates/layouts/main.jinja
    - edit line 9 to be `{# {% if (FEC_CMS_ENVIRONMENT == 'PRODUCTION') %} #}`
    - edit line 9 to be `{# {% endif %} #}`
    - or just delete them both together.
  - Load a [page](http://127.0.0.1:8000/data/raising-bythenumbers/) that uses the main template
  - View the page source or "inspect" the elements, your preference
  - Locate the code in the screenshots above (or cmd-f for "preconnect")
  - Should see the code shown above


Will need to adjust code and test on a real environment with GTM, GA, and DAP active in order to fully test Lighthouse's approval.

I'm **not** a fan of the difference in line spacing but that's a task for another ticket
____
